Sources/Common/Background Queue/QueueRunRequest.swift

```diff
@@ -131,26 +128,16 @@ public class CioQueueRunRequest: QueueRunRequest {
                     )
                 }
 
-                return self.goToNextTask(
-                    queueInventory: queueInventory,
-                    queryTotalNumberTasks: queryTotalNumberTasks,
-                    lastFailedTask: nextTaskToRunInventoryItem
-                )
+                return goToNextTask(didTaskFail: true)
             }
         }
     }
 
-    private func goToNextTask(
-        queueInventory: [QueueTaskMetadata],
-        queryTotalNumberTasks: Int,
-        lastFailedTask: QueueTaskMetadata?
-    ) {
-        var newInventory = queueInventory
-        newInventory.removeFirst()
-        runTasks(
-            queueInventory: newInventory,
-            queryTotalNumberTasks: queryTotalNumberTasks,
-            lastFailedTask: lastFailedTask
-        )
+    private func doneRunning() {
+        logger.debug("queue out of tasks to run.")
```

Comment:
Ah, I see! And this answers my query I added [here](https://github.com/customerio/customerio-ios/pull/220/files#r1033445198)